FrameworkSystem/Consumer/DynamicMonitoringConsumer.py

```diff
@@ -0,0 +1,66 @@
+"""
+Consumer in charge of reading from RabbitMQ and sending the information to Elasticsearch
+"""
+
+from dateutil import parser
+import elasticsearch, time
+from DIRAC import gLogger, gConfig, S_OK, S_ERROR
+from DIRAC.Core.Utilities.RabbitMQ import RabbitConnection
+from DIRAC.FrameworkSystem.DB.DynamicMonitoringDB import DynamicMonitoringDB
+
+class DynamicMonitoringConsumer ( object ):
+  """
+  Consumer class
+  """
+
+  def __init__( self ):
+    self.rabbitMQ = RabbitConnection()
+
+    self.docType = 'ComponentMonitoring'
+
+    # ElasticSearch initialization
+    self.elasticDB = DynamicMonitoringDB()
+
+  def start( self ):
+    """
+    Reads messages from RabbitMQ and sends them to ElasticSearch
+    """
+
+    # Read RabbitMQ parameters
+    result = gConfig.getOption( 'DIRAC/Setup' )
+    if not result[ 'OK' ]:
+      return result
+    setup = result[ 'Value' ]
+    result = gConfig.getOption( 'DIRAC/Setups/%s/%s' % ( setup, system ) )
+    if not result[ 'OK' ]:
+      return result
+    sysSetup = result[ 'Value' ]
+    result = gConfig.getOptionsDict( 'Systems/%s/%s/MessageQueueing/%s' % ( system, sysSetup, queueName ) )
+    if not result[ 'OK' ]:
+      return result
+    parameters = result[ 'Value' ]
```

Comment:
These lines above should be generalized, otherwise every consumer would end up doing the same.